Update the docs handbook from week 12 onwards (#3020)
kubernetes/sig-release: release-team/role-handbooks/docs/Release-Timeline.md

    - [Update Slack](#update-slack)
    - [Reassign issues](#reassign-issues)
    - [Clean up access](#clean-up-access)

>   - [Review Docs Process & Transition](#review-docs-process--transition)

Reviewer: This link is broken

This PR can be merged on release day by the Docs lead.

## Release Week (Week 12)

> [!Important]
> ⚠️ Everything in this section is important. It's OK to ask for advice if you're not sure.

> #### Coordinate with Branch Release Management

Reviewer: ```suggestion
### Coordinate with Branch Release Management
```

> [!Important]

> > Coordinate with the Branch Release Management team regarding the exact release cut date and time. The Final Release Notes PR must be merged at least a day before the release cut. Failure to do so will result in an outdated CHANGELOG file in the `k/k` repo.

Reviewer: ```suggestion
> Coordinate with the Branch Release Management team regarding the exact release cut date and time. The Final Release Notes PR must be merged at least a day before the release cut. Failure to do so will result in an outdated CHANGELOG file in the `kubernetes/kubernetes` repo.
```

The release cut happens quite early on the release day, a couple of hours before the Release Day Party. The Docs Lead needs to coordinate with the Branch Release Management team on the release cut timing in order to prepare the Final Release Notes PR, which must be merged before the release cut. As the Final Release Notes PR requires approval from multiple SIGs, starting the work a couple of weeks early is recommended - make sure the PR has the `lgtm` and the `approved` labels at least a day before merge.

### Update the site configuration files for previous releases

Working from your fork of the `kubernetes/website` repository, update the `config.toml` files for the 4 previous releases. You need to open 4 separate PRs, because each release has its own `release-` branch.

For the immediately previous release, there will not be a `release-[previous-release]` branch, so you should open the pull request updating the site configuration for the previous release targeting `main` and update the base branch to `release-[previous-release]` once you have created the `release-[previous-release]` branch.

> [!Warning]

> > ⚠️ DO NOT MERGE **ANY** OF THE CONFIGURATION PULL REQUESTS UNTIL THE RELEASE HAS OCCURRED

Reviewer: ```suggestion
> DO NOT MERGE **ANY** OF THE CONFIGURATION PULL REQUESTS UNTIL THE RELEASE HAS OCCURRED
```
Warning callout already has the ⚠️ icon

See this for example (1.29 was the "future release"):
* 1.25 https://github.com/kubernetes/website/pull/44302
* 1.26 https://github.com/kubernetes/website/pull/44301
* 1.27 https://github.com/kubernetes/website/pull/44299
* 1.28 https://github.com/kubernetes/website/pull/44298

Changes required:

> - Set `baseURL` to versioned docs URL i.e `https://v[current-release-hyphenated].docs.kubernetes.io/`. `[current-release-hyphenated]` is the release version with dots replaced by hyphens (e.g. `1.35` -> `1-35`). 

Reviewer: Your previous PR's also contain changes to the 1st line baseURL
- https://github.com/kubernetes/website/pull/55468/files#diff-87c53e762995c279bfb9cba355186bc80fdf748d4cd29b3898d06368d7eeb99fR1
- https://github.com/kubernetes/website/pull/55450/files#diff-87c53e762995c279bfb9cba355186bc80fdf748d4cd29b3898d06368d7eeb99fR1

But none of the above mentioned v1.25 to v1.28 contain them
Should we also update the above PRs with the new v1.35 or v1.36 cycles?

Author: Correct, this was because it was not available in the previous releases. Going forward, since we will be doing this whenever a new release is added, the changes will be retained, so there is no need to add them for the previous releases.

git push origin update-release-1.27-hugo.toml

> In this case, a PR is submitted against `release-1.27` from the fork's branch `update-release-1.27-hugo.toml`. 

Reviewer: ```suggestion
# In this case, a PR is submitted against `release-1.27` from the fork's branch `update-release-1.27-hugo.toml`. 
```

```
> [!Warning]

> > ⚠️ DO NOT MERGE **ANY** OF THE CONFIGURATION PULL REQUESTS UNTIL THE RELEASE HAS OCCURRED

Reviewer: ```suggestion
> DO NOT MERGE **ANY** OF THE CONFIGURATION PULL REQUESTS UNTIL THE RELEASE HAS OCCURRED
```

Also this comment is replicated at L684-685, is that intentional or it's okay to keep only one?

Author: This was intentional.

### Update the site configuration files for future release

The future release branch site configuration also needs to be updated to reflect the latest patch versions of the past releases.

> Create a PR against the `dev-[future-release]` branch to update the `config.toml` file that matches the [PRs for previous releases] (#update-the-site-configuration-files-for-previous-releases).

Reviewer: ```suggestion
Create a PR against the `dev-[future-release]` branch to update the `config.toml` file that matches the [PRs for previous releases](#update-the-site-configuration-files-for-previous-releases).
```

Extra space breaks the hyperlink here

### Inform localization teams

> Let localization teams know about the `k/website` repo freeze and the tentative timeline(s) for important dates with a comment to the GitHub discussion created earlier in the release cycle.

Reviewer: ```suggestion
Let localization teams know about the `kubernetes/website` repo freeze and the tentative timeline(s) for important dates with a comment to the GitHub discussion created earlier in the release cycle.
```

**Template:**
```
Hello localization team leads! 

I don't think any action is required from you, but I wanted to let you know that we are on track for the release on [Date of Release] 
and all Kubernetes website branches are up to date (main, dev-[future release]). 

> Let me know if I can help with anything! Thanks!

Reviewer: ```suggestion
Hello localization team leads!

I don't think any action is required from you, but I wanted to let you know that we are on track for the release on [Date of Release] 
and all Kubernetes website branches are up to date (main, dev-[future release]).

Let me know if I can help with anything! Thanks!
```

  - e.g, site name: k8s-v1-20
  - e.g, custom domain: v1-20.docs.kubernetes.io

> [!Note]

> >  if the `release-[current-release]` is created before the website is frozen you may need to merge in main to keep up-to-date

Reviewer: ```suggestion
> If the `release-[current-release]` is created before the website is frozen you may need to merge in main to keep up-to-date
```

Now create a pull request to merge the new branch you've made into the `release-[current-release]` branch on [k/website](https://github.com/kubernetes/website).

> [!Note]
> Make a note of the commit hash of the last commit of the `[release-current-release]` branch as it will be used to tag the commit as the final commit for the snapshot.

#### Merge `main`

After the freeze, if the dev-[future-release] branch is behind `main`, create a PR to merge `main` into dev-[future-release].

[Sync](#weekly-branch-sync-prs) `main` into the release-[current-release] branch the day before the release.

After review from SIG Docs, both PRs will need to be merged manually using the `Create a merge commit` method of merging.

#### Get approvals for open PRs

At this stage, [integration branch] PR and configuration PRs should be ready to be reviewed and approved by SIG Docs.
Request a review for all open PRs, especially the [integration branch], to make sure it can be merged on time during the
release day.

> ### Publish final Release Notes

Reviewer: ```suggestion
#### Publish final Release Notes
```

- Final version of release notes committed for release.
- Close the _Known Issues_ Issue and make sure everything has been resolved.
- Release Notes must be merged into `master` prior to the release. If this is not done the release will include the latest draft.

> - Keep an eye on the `#release-notes` channel for any requests for any questions, edits or missed release notes.

Reviewer: ```suggestion
- Keep an eye on the `#release-notes` channel for any questions, requests, edits, or missed release notes.
```

#### Review milestone

Review milestone for completion and outstanding PRs. For PRs that won't make the release, change their milestone and make sure everyone is clear.

## Release Day

> [!Important]
> ⚠️ Everything in this section is important. It's OK to ask for advice if you're not sure.

This process takes approximately 4 hours.

Coordinate with the Branch Release Management, SIG-Docs Chair, and Release Lead for the exact timing of each step. Typically, after the release is 'officially' built, you proceed to merge the docs, and then you approve the blog post to "make it official". For 1.21, we merged docs at 11:00 am PDT and the blog was merged at 11:30 am PDT - marking the release "complete".
    
### Merge the integration branch

> Once release management team has successfully cut the release, Docs Lead or the coordinating SIG-Docs Chair will merge the [integration branch] manually using the `Create a merge commit` method of merging. Do not delete the dev-[future-release] when GitHub asks.

Reviewer: ```suggestion
Once the release management team has successfully cut the release, Docs Lead or the coordinating SIG-Docs Chair will merge the [integration branch] manually using the `Create a merge commit` method of merging. Do not delete the dev-[future-release] when GitHub asks.
```

  - The current lead must stay in the website milestone maintainers list until the `[current release]` website version is officially supported.
  - The future lead must stay in the website milestone maintainers list to prepare for the next release.
- ⚠️  Create a PR against the **[kubernetes/website](https://github.com/kubernetes/website) repo** to **remove** the current Docs Lead from [sig-docs-en-owners](https://github.com/orgs/kubernetes/teams/sig-docs-en-owners)

### Review Docs Process, Update Handbook and Transition

#### Review & Updates

Once the release is over, it is highly recommended to go over the handbook and identify areas of improvement. The Docs Lead is responsible for holding a docs-only retro with the SIG-Docs team. The Release Retro meetings can also be used to discuss process and handbook improvements with the broader team. Ask the shadows to share their thoughts and suggestions as well.

The questions to ask in the retro meetings:
- Can we improve the process? The challenges that were faced during the release.

> - Can we improve the playbook? The areas handbook misses to guide the team.

Reviewer: ```suggestion
- Can we improve the playbook? The areas where the handbook fails to guide the team.
```

- Can we add or **remove** anything from the playbook?

#### Transition to the Next Lead

> Once the handbook is updated, conduct a transition call with the incoming Lead. Walk through the process, your experience, and your learnings. Encourage them to ask questions. You can also go through the individual sections of the handbook to explain what happened at each step. Remember, the primary goal of the meeting is to walk them through the steps that only the Docs Lead performs - particularly during the last week of the release and on release day itself.

Reviewer: ```suggestion
Conduct a transition call with the incoming Lead. Walk through the process, your experience, and your learnings. Encourage them to ask questions. You can also go through the individual sections of the handbook to explain what happened at each step. Remember, the primary goal of the meeting is to walk them through the steps that only the Docs Lead performs - particularly during the last week of the release and on release day itself.
```

Incase there aren't any updates to the handbook, we should still probably have a transition call
